### ISSUE TITLE
AV program detects malware in windows download of DMD

### ISSUE THREAD

**Issue by dlangBugzillaToGithub** [dlang/installer#631]
### CC List

* Martin Nowak (MartinNowak)
* Alexey Kulentsov
* Seb
* Mike Franklin
* David M

### Description

````markdown
Created attachment 1687
Output of AV program

When I ran DMD download (dmd-2.063.2) on the Virus total site I got the following 2 detections

McAfee-GW-Edition

BehavesLike.Win32.Dropper.cc

VBA32

suspected of Trojan.Downloader.gen.h

Is this correct or false positives.  If correct can you direct me to a clean version

Thanks
````

**!!!There are attachements in the bugzilla issue that have not been copied over!!!**

**Comment by dlangBugzillaToGithub** [comment]
#### greeenify commented on 2018-04-20T20:56:27Z



````markdown
This is a false positive. Please notify your Antivirus vendor and report their false detection there. Thanks!

BTW dmd-2.063 is more than four years old. Are you sure you need such an old release?
````

**Comment by dlangBugzillaToGithub** [comment]
#### vintagedave commented on 2018-05-24T21:41:44Z



````markdown
dmd-2.080.0.exe, downloaded yesterday, gives 12 reports on VirusTotal including McAfee, TrendMicro, and Microsoft.

This is 12/64 scanners, ie 18%.  A false positive sounds less likely.

https://www.virustotal.com/#/file/007560cc35e78ba74d6fa9732e27032a1fd2f2d6cbadffd1c39ffff68d5dd100/detection

Windows Defender on Win10 identifies it as a trojan and will not run, too.
````

**Comment by dlangBugzillaToGithub** [comment]
#### greeenify commented on 2018-05-24T22:09:43Z



````markdown
It's a false positive. You can check the signature of the binary.

Please report it to your Antivirus vendors. They traditionally have troubles with the DigitalMars runtime.
````

**Comment by dlangBugzillaToGithub** [comment]
#### greeenify commented on 2018-05-25T05:37:55Z



````markdown
> I do not believe responsibility for reporting a false positive

Well, you are the one using the snake oil software (and possibly even paying for it).

Don't forget that D is an open source project and driven by volunteers. Most D developers use Linux, so they never run into this problems with Windows.
The only thing I can guarantee you is that it's a false positive because these reports have been semi-regularily coming in from time to time over the recent years. As mentioned for the AV vendors the D runtime looks still unfamiliar and thus they often wrongly determine it to be a virus.

So tl;dr: if you don't report it to the AV vendor you use, who else is going to?
And also AV vendors often take reports from their users much more seriously than from open source projects (I tried to get in touch with done of them a few years ago which horribly failed).
````

**Comment by dlangBugzillaToGithub** [comment]
#### slavo5150 commented on 2018-05-25T05:56:23Z



````markdown
There is something screwy about it.  It's not the compiler that is reporting the virus, it's the installer.  What utility are we using the generate the installer executable?
````

**Comment by dlangBugzillaToGithub** [comment]
#### code commented on 2018-11-01T03:00:39Z



````markdown
(In reply to Alexey Kulentsov from comment #10)
> DMD 2.082.0, Microsoft Windows 10 Enterprise Build 17134 - Defender prevents
> any work with DMD or compiled by DMD programs.

We're actually signining Windows binaries since 2.082.0 and didn't have any further Defender issues after also submitting dmd for inspection.

Do you have any additional info to reproduce the issue, in particular the one with self-built binaries.

[¹]: https://dlang.org/changelog/2.082.0.html#signed_windows_binaries
````

**Comment by dlangBugzillaToGithub** [comment]
#### dfj1esp02 commented on 2018-11-01T11:30:05Z



````markdown
I downloaded 2.082.1, Edge showed message "Running security scan" which took a long time, I opened the Defender interface, and after some time Edge reported that download is complete. Maybe the scan just takes a long time.
````

**Comment by dlangBugzillaToGithub** [comment]
#### crimaniak commented on 2018-11-09T13:19:41Z



````markdown
Created attachment 1716
Compiled utility rejected by Windows Defender in corporate network
````

**Comment by dlangBugzillaToGithub** [comment]
#### crimaniak commented on 2018-11-09T13:25:16Z



````markdown
(In reply to Martin Nowak from comment #11)
> We're actually signining Windows binaries since 2.082.0 and didn't have any
> further Defender issues after also submitting dmd for inspection.
I am in the corporate network with centralized administration. Now I think, maybe, corporate rules are more strict than defaults, and rejects unknown self-signed certificates.

> Do you have any additional info to reproduce the issue, in particular the
> one with self-built binaries.

DMD installer from official site, DMD version:
DMD32 D Compiler v2.082.0

Copyright (C) 1999-2018 by The D Language Foundation, All Rights Reserved written by Walter Bright

Utility source and executable (mpgit) are attached to the issue in the previous comment.
````

**Comment by dlangBugzillaToGithub** [comment]
#### greeenify commented on 2018-11-10T01:13:54Z



````markdown
It's not a self-Signet certificate, it's one from an officially approved windows certificate vendor. Not sure whether we can do more for you. Maybe you just need to talk to your admins?
````
